src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -34,10 +36,13 @@ internal partial class DiagnosticIncrementalAnalyzer
             CompilerDiagnosticsScope compilerDiagnosticsScope,
             bool isActiveDocument, bool isVisibleDocument,
             bool isOpenDocument, bool isGeneratedRazorDocument,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken,
+            out bool skipAnalysisForNonCachedDocument)
```

Comment:
nit - maybe a more clear name would be isAnalyzerEnabled / isAnalyzerSuppressed?

The current name is a little hard to parse (skip...noncached)